py-rse/package-py.Rmd

```diff
@@ -482,450 +173,562 @@ without affecting the main installation (Figure \@ref(fig:py-rse-package-py-vir
+```
 
-People can always get your package by cloning your repository and copying files from that
-(assuming your repository is accessible,
-which is should be for published research),
-but it's much friendlier to create something they can install
-(and it will be easier for us when it comes to running test suites
-and distributing, too!)
+```shell
+$ pip install -e .
+```
 
-For historical reasons,
-Python has several ways to build an installable package.
-We will show how to use [setuptools][setuptools],
-which is the lowest common denominator;
-[conda][conda] is a modern does-everything solution,
-but the setuptools approach we describe below will
-allow everyone, regardless of what python distribution
-they use, to easily use your package.
+```text
+Defaulting to user installation because normal site-packages is not writeable
+Obtaining file:///home/amira/zipfs_law
+Requirement already satisfied: matplotlib in /usr/lib/python3.8/site-packages (from zipfs-law==0.1) (3.2.1)
+Requirement already satisfied: pandas in /home/amira/.local/lib/python3.8/site-packages (from zipfs-law==0.1) (1.0.3)
+Requirement already satisfied: scipy in /usr/lib/python3.8/site-packages (from zipfs-law==0.1) (1.4.1)
+Requirement already satisfied: pyyaml in /usr/lib/python3.8/site-packages (from zipfs-law==0.1) (5.3.1)
+Requirement already satisfied: cycler>=0.10 in /usr/lib/python3.8/site-packages (from matplotlib->zipfs-law==0.1) (0.10.0)
+Requirement already satisfied: kiwisolver>=1.0.1 in /usr/lib/python3.8/site-packages (from matplotlib->zipfs-law==0.1) (1.1.0)
+Requirement already satisfied: numpy>=1.11 in /usr/lib/python3.8/site-packages (from matplotlib->zipfs-law==0.1) (1.18.2)
+Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 in /usr/lib/python3.8/site-packages (from matplotlib->zipfs-law==0.1) (2.4.6)
+Requirement already satisfied: python-dateutil>=2.1 in /usr/lib/python3.8/site-packages (from matplotlib->zipfs-law==0.1) (2.8.1)
+Requirement already satisfied: pytz>=2017.2 in /usr/lib/python3.8/site-packages (from pandas->zipfs-law==0.1) (2019.3)
+Requirement already satisfied: six in /usr/lib/python3.8/site-packages (from cycler>=0.10->matplotlib->zipfs-law==0.1) (1.14.0)
+Requirement already satisfied: setuptools in /usr/lib/python3.8/site-packages (from kiwisolver>=1.0.1->matplotlib->zipfs-law==0.1) (46.1.3)
+Installing collected packages: zipfs-law
+  Running setup.py develop for zipfs-law
+Successfully installed zipfs-law
+```
+
+The output looks slightly different this time
+because pip did not need to go online to fetch the repositories,
+since there was a local cached version from the previous install.
+
+Now we can use our commands directly from the terminal
+without writing the full path to the file
+and without prefixing it with `python`.
 
-To use `setuptools`,
-we must create a file called `setup.py` in the directory *above* the root directory of the package:
+```shell
+countwords ../data/dracula.csv
```

Comment:
Instead of using a `...` when displaying the output you could call: `countwords ../data/dracula.csv -n 5`